mbq/atomiq/consumers.py

```diff
@@ -40,46 +27,26 @@ def process_task(self, task):
                 task.failed_at = arrow.utcnow().datetime
                 task.error_message = str(e)
                 task.stacktrace = traceback.format_exc()
-                rollbar.report_exc_info()
             else:
                 # Task will be retried. It will be put back on the queue and made
                 # invisible to the consumer using an exponential backoff policy.
                 backoff_time = 2**task.number_of_attempts
                 task.visible_after = arrow.utcnow().shift(seconds=backoff_time).datetime
         else:
-            # Task execution succeeded.
             task.state = constants.TaskStates.SUCCEEDED
             task.succeeded_at = arrow.utcnow().datetime
 
         task.save()
 
-    def send_task_execution_metrics(self, task, execution_started_at, execution_ended_at):
-        dd_tags = {
-            'end_state': task.state,
-            'result': 'success' if task.state == constants.TaskStates.SUCCEEDED else 'error',
-            'queue_type': self.queue_type,
-        }
-        _collector.increment(
-            'task',
-            tags=dd_tags,
-        )
+        return task
 
-        _collector.timing(
-            'task.wait_time_ms',
-            utils.time_difference_ms(task.visible_after, execution_started_at),
-            tags=dd_tags,
-        )
-        _collector.timing(
-            'task.execution_time_ms',
-            utils.time_difference_ms(execution_started_at, execution_ended_at),
-            tags=dd_tags,
-        )
-        if task.state == constants.TaskStates.SUCCEEDED:
-            _collector.timing(
-                'task.turnaround_time_ms',
-                utils.time_difference_ms(task.created_at, task.succeeded_at),
-                tags=dd_tags,
-            )
+    def get_next_enqueued_task(self):
+        try:
+            task = self.model.objects.available_for_processing()[:1].select_for_update().get()
+        except self.model.DoesNotExist:
+            raise exceptions.NoAvailableTasksToProcess()
```

Comment:
nab: Now that you're raising your own descriptive exception (which is good) we can actaully simplify this `get_next_enqueued_task` function:

```
task = self.model.objects.available_for_processing().select_for_update().first()
if not task:
    raise exceptions.NoAvailableTasksToProcess()
```